### ISSUE TITLE
new rule - useless `@final` decorator on private method

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#24221]
### Summary

Originally raised at https://github.com/astral-sh/ty/issues/3140

### `main.py`

```py
from typing import final, override


class Foo:
    @final # no error. should say something like "@final decorator is useless on private methods"
    def __foo(self):
        ...

class Bar(Foo):
    @override
    def __foo(self): # correct error: Method `__foo` is decorated with `@override` but does not override anything
        ...
```

**Comment by ntBre** [comment]
cc @AlexWaygood 

It's not really clear to me from the ty discussion if Alex and Charlie thought this would be a better fit for Ruff, or if it doesn't seem worth linting against at all. Have you seen this pattern in a real project where the rule would have been helpful? I'm just trying to get a sense for how common this is. It seems quite niche, but I guess I'm not really opposed to it otherwise.
